### DISCUSSION TITLE
pocketbase JS SDK: getFullList() batch option for what?

### DISCUSSION THREAD

**Discussion by ramadhanafif** [pocketbase/pocketbase#4677] [Q&A] [answered]
On using `getFullList()` method, I noticed the `batch` options.
What is it used for? and what is the difference between batch and perPage?

Assumes that I have a database with 1000 record. The `getFullList()` by default take 500 perPage per queries, and the batch option means I can get the whole 1000 record in two batches, is this understanding correct?

Example code:
```js
const res = await pb.collection("users").getFullList({
	batchSize: 2
});

```

**Comment by ganigeorgiev** [comment, marked as answer]
> Assumes that I have a database with 1000 record. The getFullList() by default take 500 perPage per queries, and the batch option means I can get the whole 1000 record in two batches, is this understanding correct?

~Yes.~ Actually on second read of your code sample - No, your example is incorrect. The `batch` specify how many items per batch to fetch.

> What is it used for? and what is the difference between batch and perPage?

`getFullList()` sends `getList()` requests on batches until all items are fetched.

`batch` (_note that there is no `batchSize`_) and `perPage` technically are the same in this context. The `batch` query parameter exists primarily for backward compatibility with earlier versions of the SDK and it predates the `SendOptions` support added with [JS SDK v0.17](https://github.com/pocketbase/js-sdk/releases/tag/v0.17.0) (_the old syntax `getFullList(batch, options)` should still work_).

In the future this may get cleanup a little to avoid such confusions because currently the JS SDK is littered with functions overloads.

**Reply by ramadhanafif** [reply]
Appreciate the response, thanks!

**Comment by Psypher1** [comment]
I have a folow up question: How do I limit `getFullList` to a specific number of entries?

This isn't working for me:

```js
const res = await pb.collection("links").getFullList(5, {
	sort: "-clicks"
});
```

**Reply by ganigeorgiev** [reply]
@Psypher1 I'm not sure I follow and I don't understand how your question is related to the original one.

In any case I don't recommend setting `getFullList` batch size to 5 as this will send a lot of requests if you have many records. Note `getFullList()` works by calling `getList()` recursively until **all** records are fetched.

If you just want to fetch **only 5** records, then use `getList`:

```js
const res = await pb.collection("links").getList(1, 5, {
    sort: "-clicks",
    skipTotal: true, // optional if you don't need the total* count fields
})
console.log(res)
```

If that's not what you are looking for, then please don't hijack old questions and instead open a new Q&A discussion with clear details about your use case and the problem you are trying to solve.

**Reply by Psypher1** [reply]
@ganigeorgiev Thank you for explaining this. 

I do apologise if it seemed like I hijacked the question. In my search, this was the closest to what I am trying to do. I have been reprimanded for asking new questions that may have already solved.

I will ask new questions in future
